sympy/physics/continuum_mechanics/column.py

```diff
@@ -0,0 +1,323 @@
+
+
+    def solve_slope_deflection(self):
+        """
+        Solves the differnetial equation of buckling to determine the
+        deflection and slope equations.
+
+        Examples
+        ========
+
+        A column of timber section is 8m long both ends being fixed.
+        Young's modulus of the timber is E and second-moment I.
+        The applied load on the column is 360KN.
+
+        >>> from sympy.physics.continuum_mechanics.column import Column
+        >>> from sympy import Symbol, symbols
+        >>> E, I, P = symbols('E, I, P', positive=True)
+        >>> c = Column(8, E, I, 360000, top="fixed", bottom="fixed")
+        >>> c.end_conditions
+        {'bottom': 'fixed', 'top': 'fixed'}
+        >>> c.boundary_conditions
+        {'deflection': [(0, 0), (8, 0)], 'slope': [(0, 0)]}
+        >>> c.moment()
+        -M + 360000*y(x)
+        >>> c.solve_slope_deflection()
+        >>> c.deflection()
+        -M*cos(600*x/(sqrt(E)*sqrt(I)))/360000 + M/360000
+        >>> c.slope()
+        M*sin(600*x/(sqrt(E)*sqrt(I)))/(600*sqrt(E)*sqrt(I))
+        """
+        y = Function('y')
+        x = self._variable
+        P = Symbol('P', positive=True)
+
+        C1, C2 = symbols('C1, C2')
+        E = self._elastic_modulus
+        I = self._second_moment
+
+        # differnetial equation of Column buckling
+        eq = E*I*y(x).diff(x, 2) + self._moment
+
+        self._deflection = dsolve(Eq(eq, 0), y(x), ics={y(0): 0, y(x).diff(x).subs(x, 0): 0}).args[1]
+
+        # if deflection is zero, no buckling occurs, which is not the case,
+        # so trying to solve for the constants differently
+        if self._deflection == 0:
+            self._deflection = dsolve(Eq(eq, 0), y(x)).args[1]
+
+            defl_eqs = []
+            # taking last two bounndary conditions which are actually
+            # the initial boundary conditions.
+            for point, value in self._boundary_conditions['deflection'][-2:]:
+                defl_eqs.append(self._deflection.subs(x, point) - value)
+
+            # solve for C1, C2 along with P
+            solns = solve(defl_eqs, (P, C1, C2), dict=True)
+            for sol in solns:
+                if self._deflection.subs(sol) == 0:
+                    # removing trivial solutions
+                    solns.remove(sol)
+
+            # checking if the constants are solved, and subtituting them in
+            # the deflection and slope equation
+            if C1 in solns[0].keys():
+                self._deflection = self._deflection.subs(C1, solns[0][C1])
+            if C2 in solns[0].keys():
+                self._deflection = self._deflection.subs(C2, solns[0][C2])
+            if P in solns[0].keys():
+                self._critical_load = solns[0][P]
+
+        self._slope = self._deflection.diff(x)
+
+
+    def critical_load(self):
+        """
+        Detrmines the critical load (for single bow buckling condition) of
```

Comment:
Here as well, `Determines`.

Reply:
I have mentioned this for my GSoC proposal. I want to continue this module for other loading cases .